src/neuroconv/tools/spikeinterface/spikeinterface.py

```diff
@@ -318,14 +321,14 @@ def add_electrodes(
     # Channel name logic
     channel_ids = checked_recording.get_channel_ids()
     if "channel_name" in data_to_add:
+        # if 'channel_name' is set as a property, it is used to override default channel_ids (and "id")
         channel_name_array = data_to_add["channel_name"]["data"]
     else:
         channel_name_array = channel_ids.astype("str", copy=False)
         data_to_add["channel_name"].update(description="unique channel reference", data=channel_name_array, index=False)
-
-    # If the channel ids are integer keep the old behavior of asigning nwbfile.electrodes.id equal to channel_ids
-    if np.issubdtype(channel_ids.dtype, np.integer):
-        data_to_add["id"].update(data=channel_ids, index=False)
+        # If the channel ids are integer keep the old behavior of asigning nwbfile.electrodes.id equal to channel_ids
+        if np.issubdtype(channel_ids.dtype, np.integer):
+            data_to_add["id"].update(data=channel_ids, index=False)
```

Comment:
@CodyCBakerPhD as discussed yesterday, passing the `channel_name` property overrides the default names assigned by `neuroconv`. However, if `channel_ids` are `int`, the `id` field of the rows is not consistent. In this case, we only set the `id` if the `channel_name` property is not set.